### PR TITLE
fix(vrl): Fix naming of `compact`'s `object` argument

### PR DESCRIPTION
**Pull request by jszwedko**
It was implemented as `map`, but was documented as `object`, which is
how we refer to key/value types in VRL now (previously we did use
`map`).

Fixes: #9958

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: lib/vrl/stdlib/src/compact.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ impl Function for Compact {
         let recursive = arguments.optional("recursive");
         let null = arguments.optional("null");
         let string = arguments.optional("string");
-        let map = arguments.optional("map");
+        let  object = arguments.optional("object");
```
Comment (spencergilbert) [inline lib/vrl/stdlib/src/compact.rs:78]: Extra space?

**Review by spencergilbert — APPROVED** [review]
Also noting as a breaking change?

File: lib/vrl/stdlib/src/compact.rs (hunk 1 above)
Comment (jszwedko) [reply lib/vrl/stdlib/src/compact.rs:78]: Oh, yeah, weird, `rustfmt` didn't catch that.

**Comment by jszwedko** [comment]
> Also noting as a breaking change?

Yeah, I thought about that, but I think this is a bug given that it is inconsistent with the rest of VRL and didn't match the docs. Granted, there may be users that switched to using `map` based on the error feedback. I think it's right on the line and would be inclined just to call it out in the release notes, but not as a breaking change.

**Review by JeanMertz — APPROVED** [review]
It's also interesting to consider what _is_ a "breaking change" in Vector's case.

Personally, I'd consider anything that can _break your runtime expectations_ to be breaking changes, and don't consider configuration changes that prevent Vector from booting to be breaking changes.

In VRL's case, these sorts of changes caused the compiler to fail at boot, so I don't consider them breaking changes either, even though it requires operators to put in a bit of work to upgrade their programs.

Obviously, it's contrived, and others probably think differently.

**Comment by jszwedko** [comment]
> Personally, I'd consider anything that can _break your runtime expectations_ to be breaking changes, and don't consider configuration changes that prevent Vector from booting to be breaking changes.
> 
> In VRL's case, these sorts of changes caused the compiler to fail at boot, so I don't consider them breaking changes either, even though it requires operators to put in a bit of work to upgrade their programs.

Yeah, it's a bit of an art. I think incompatible configuration changes are breaking changes, but that they have have less of an impact or risk given that they are very visible and don't result in, for example, silent behavior changes.

There is an extreme stance that all bug fixes are breaking changes to those relying on the bug behavior, but I think that would make release notes overly noisy. I think it has to be balanced with the percentage of users likely to have been depending on the bug behavior.

**Pull request merged**